apps/survey/contracts/Surveying.sol

```diff
@@ -0,0 +1,240 @@
+    function changeMinAcceptParticipationPct(uint256 _minAcceptParticipationPct) authP(MODIFY_PARTICIPATION_ROLE, arr(_minAcceptParticipationPct)) external {
+        require(_minAcceptParticipationPct > 0);
+        require(_minAcceptParticipationPct <= PCT_BASE);
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+
+        ChangeMinParticipation(_minAcceptParticipationPct);
+    }
+
+    /**
+    * @notice Create a new non-binding survey about "`_metadata`"
+    * @param _metadata Survey metadata
+    * @param _options Number of options voters can decide between
+    * @return surveyId id for newly created survey
+    */
+    function newSurvey(string _metadata, uint256 _options) auth(CREATE_SURVEYS_ROLE) external returns (uint256 surveyId) {
+        return _newSurvey(_metadata, _options);
+    }
+
+    /**
+    * @notice Vote for multiple options in survey #`_surveyId`.
+    * @dev Empty options and stakes arrays allow to remove previous votes
+    * @param _surveyId Id for survey
+    * @param _options Array with indexes of supported options
+    * @param _stakes Number of tokens assigned to each option
+    */
+    function voteOptions(uint256 _surveyId, uint256[] _options, uint256[] _stakes) public {
+        require(_options.length == _stakes.length);
+        require(canVote(_surveyId, msg.sender));
+
+        uint256 i;
+        Survey storage survey = surveys[_surveyId];
+
+        uint256 voterStake = token.balanceOfAt(msg.sender, survey.snapshotBlock);
+        MultiOptionVote storage previouslyVoted = survey.voters[msg.sender];
+
+        // revert previous votes, if any
+        if (previouslyVoted.options.length > 0) {
+            // voter removes their vote
+            for (i = 1; i < previouslyVoted.options.length; i++) {
+                survey.votes[previouslyVoted.options[i]] = survey.votes[previouslyVoted.options[i]].sub(previouslyVoted.stakes[i]);
+            }
+            // remove previous vote participation
+            survey.participation = survey.participation.sub(voterStake.sub(previouslyVoted.stakes[0]));
+        }
+
+        // reset previously voted options, if any
+        delete survey.voters[msg.sender];
+
+        uint256 totalVoted = 0;
+        // reserve first index for NO_VOTE
+        survey.voters[msg.sender].options.push(NO_VOTE);
+        survey.voters[msg.sender].stakes.push(0);
+        for (i = 0; i < _options.length; i++) {
+            _voteOption(_surveyId, _options[i], _stakes[i]);
+            // let's avoid repeating an option by
+            // making sure that ascending order is preserved in options array
+            require(survey.voters[msg.sender].options[i] < survey.voters[msg.sender].options[i+1]);
+            totalVoted = totalVoted.add(_stakes[i]);
+        }
+
+        // compute and register non used tokens
+        // implictly we are doing require(totalVoted <= voterStake) too:
+        survey.voters[msg.sender].stakes[0] = voterStake.sub(totalVoted);
```

Comment:
Not sure what you mean by "magic number" but I wouldn't say this is one. As stated in a previous comment (and in your next one), first index (`0`) for `options` and `stakes` arrays is reserved for `NO_VOTE`, but `0` here means the index, not the value. Although it certainly work too, I would find confusing using `NO_VOTE` here instead of `0`.